src/Parser.js

```diff
@@ -137,31 +142,25 @@ Parser.prototype.parseExpression = function(breakOnInfix, breakOnToken) {
     // we reached the end, a }, or a \right)
     while (true) {
         var lex = this.nextToken;
-        var pos = this.pos;
         if (endOfExpression.indexOf(lex.text) !== -1) {
             break;
         }
         if (breakOnToken && lex.text === breakOnToken) {
             break;
         }
+        if (breakOnInfix && functions[lex.text] && functions[lex.text].infix) {
+            break;
+        }
```

Comment:
This is the replacement for the rewininding which I deleted a few lines down. Before this change, if the next *atom* was of type infix, then we rewound the input stream to parse everything a second time. Now I'd like to concentrate on linear operations, with no jumping around in the input stream. Thus we have to detect an infix token before it's parsed to an atom. Thus the check above, and the addition of the matching metadata to the table of functions.